src/wolfcrypt/_build_ffi.py

```diff
@@ -18,14 +18,81 @@
 # along with this program; if not, write to the Free Software
 # Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA
 
+import os
 from distutils.util import get_platform
 from cffi import FFI
 from wolfcrypt import __wolfssl_version__ as version
-from wolfcrypt._build_wolfssl import local_path
+from wolfcrypt._build_wolfssl import wolfssl_inc_path, wolfssl_lib_path
 
-ffi = FFI()
+# detect features if user has built against local wolfSSL library
+# if they are not, we are controlling build options in _build_wolfssl.py
+local_wolfssl = os.environ.get("USE_LOCAL_WOLFSSL")
+if local_wolfssl is not None:
+    # open <wolfssl/options.h> header to parse for #define's
+    # This will throw a FileNotFoundError if not able to find options.h
+    optionsHeaderPath = wolfssl_inc_path() + "/wolfssl/options.h"
+    optionsHeader = open(optionsHeaderPath, 'r')
+    optionsHeaderStr = optionsHeader.read()
+    optionsHeader.close()
+else:
+    optionsHeaderStr = ""
 
-ffi.set_source(
+# default values
```

Comment:
Either the defaults are not correct, or the native feature detection is not working on my system.

wolfBoot tools work only if I change defaults:
```python
ECC_ENABLED=1
ED25519_ENABLED=1
KEYGEN_ENABLE=1
```
Is it possible to enable these features by default?

Reply:
Please note also that the tests ran on travis for this reason are only 11 out of the 21 avaialble

Reply:
Native feature detection only happens when building with the USE_LOCAL_WOLFSSL variable set, to build against a local version of wolfSSL.  Updated README to reflect this.

Reply:
Enabled ECC, Ed25519, and keygen by default

Reply:
Thanks Chris! One last thing:

With SHA384 and SHA512 disabled by default, not all tests are being run with `make test`.
I notice that enabling these by default, with `make test` we get all 23 tests (instead of just 19).

Could you please consider also:

```diff
diff --git a/src/wolfcrypt/_build_ffi.py b/src/wolfcrypt/_build_ffi.py
index fbd8f53..58ffa9c 100644
--- a/src/wolfcrypt/_build_ffi.py
+++ b/src/wolfcrypt/_build_ffi.py
@@ -50,8 +50,8 @@ else:
 MPAPI_ENABLED = 0
 SHA_ENABLED = 1
 SHA256_ENABLED = 1
-SHA384_ENABLED = 0
-SHA512_ENABLED = 0
+SHA384_ENABLED = 1
+SHA512_ENABLED = 1
 DES3_ENABLED = 1
 AES_ENABLED = 1
 HMAC_ENABLED = 1
```

Thanks!

/d

Reply:
Done, thanks Daniele!